python/sglang/srt/model_executor/cuda_graph_runner.py

```diff
@@ -710,18 +1071,42 @@ def replay_prepare(
             )
             index = bisect.bisect_left(self.capture_bs, max_batch_size)
         else:
-            index = bisect.bisect_left(self.capture_bs, raw_bs)
-        bs = self.capture_bs[index]
+            index = (
+                bisect.bisect_left(
+                    self.capture_prefill_seqlen, forward_batch.seq_lens_sum
+                )
+                if forward_batch.forward_mode == ForwardMode.EXTEND
+                else bisect.bisect_left(self.capture_bs, raw_bs)
+            )
+        bs = (
+            self.max_prefill_bs
+            if forward_batch.forward_mode == ForwardMode.EXTEND
+            else self.capture_bs[index]
+        )
+        padding_num_tokens = (
+            self.capture_prefill_seqlen[index] * bs
+            if forward_batch.forward_mode == ForwardMode.EXTEND
+            else self.num_tokens_per_bs * bs
+        )
         if bs != raw_bs:
             self.seq_lens.fill_(self.seq_len_fill_value)
             self.out_cache_loc.zero_()
 
         # Common inputs
-        self.input_ids[:raw_num_token].copy_(forward_batch.input_ids)
-        self.req_pool_indices[:raw_bs].copy_(forward_batch.req_pool_indices)
-        self.seq_lens[:raw_bs].copy_(forward_batch.seq_lens)
-        self.out_cache_loc[:raw_num_token].copy_(forward_batch.out_cache_loc)
-        self.positions[:raw_num_token].copy_(forward_batch.positions)
+        if forward_batch.forward_mode == ForwardMode.EXTEND:
+            self.prefill_input_ids[:raw_num_token].copy_(forward_batch.input_ids)
+            self.prefill_req_pool_indices[:raw_bs].copy_(forward_batch.req_pool_indices)
+            self.prefill_seq_lens[:raw_bs].copy_(forward_batch.seq_lens)
+            self.prefill_out_cache_loc[:raw_num_token].copy_(
+                forward_batch.out_cache_loc
+            )
+            self.prefill_positions[:raw_num_token].copy_(forward_batch.positions)
+        else:
+            self.input_ids[:raw_num_token].copy_(forward_batch.input_ids)
+            self.req_pool_indices[:raw_bs].copy_(forward_batch.req_pool_indices)
+            self.seq_lens[:raw_bs].copy_(forward_batch.seq_lens)
+            self.out_cache_loc[:raw_num_token].copy_(forward_batch.out_cache_loc)
+            self.positions[:raw_num_token].copy_(forward_batch.positions)
```

Comment:
(this is another example of what I said above about the "Pack" approach or the "prefill_runner=GraphRunner();decode_runner=GraphRunner()" approach)